### ISSUE TITLE
🐛 firebase_dynamic_links getInitialLink() always null

### ISSUE THREAD

**Issue by mdrideout** [firebase/flutterfire#8926]
## Bug report

**Describe the bug**
`await FirebaseDynamicLinks.instance.getInitialLink();` always returns null.

### Steps to reproduce

1. Create basic Firebase Flutter project with a test dynamic link for iOS
2. Attempt to start the non-running app by clicking a dynamic link URL
3. Application opens, but the getInitialLink() function returns null

### Expected behavior

The getInitialLink() function should return the initial link.

### Sample project
Sample shows commented out non-working `FirebaseDynamicLinks` method, and working `AppLinks` method.

```dart
import 'package:app_links/app_links.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:mbb_flutter/firebase_options.dart';

import 'src/app.dart';
import 'src/config/configure_nonweb.dart' if (dart.library.html) 'src/config/configure_web.dart';
import 'src/settings/settings_controller.dart';
import 'src/settings/settings_service.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  // Removes /#/ from URL
  configureApp();

  // Firebase: Initialize Firebase
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );

  // Firebase: Get any initial dynamic links (not compatible with web)
  // Gets passed to our [App()] which passes this to the router
  // PendingDynamicLinkData? initialLink;
  Uri? initialLink;

  if (!kIsWeb) {
    // initialLink = await FirebaseDynamicLinks.instance.getInitialLink();

    final appLinks = AppLinks();
    initialLink = await appLinks.getInitialAppLink();
  }

  // Firebase: Initialize Firebase Authentication
  await FirebaseAuth.instance.useAuthEmulator('localhost', 9099);

  // Set up the SettingsController, which will glue user settings to multiple
  // Flutter Widgets.
  final settingsController = SettingsController(SettingsService());

  // Load the user's preferred theme while the splash screen is displayed.
  // This prevents a sudden theme change when the app is first displayed.
  await settingsController.loadSettings();

  // Run the app and pass in the SettingsController. The app listens to the
  // SettingsController for changes, then passes it further down to the
  // SettingsView.
  runApp(
    // Riverpod Provider Scope
    ProviderScope(
      child: App(settingsController: settingsController, initialLink: initialLink),
    ),
  );
}
```

---

## Additional context

Essentially a skeleton Flutter project with Firebase implemented.

Using [app_links](https://pub.dev/packages/app_links) as a drop in replacement works without issue to get the initial link.

I have tried the "workaround" I found in a few places to add a delay before attempting to get the link, but that has not worked for me. 

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.4 21F79 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.67.2)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!

```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.1
Flutter SDK 3.0.1
mbb_flutter 1.0.0+1

dependencies:
- app_links 3.2.0 [flutter app_links_platform_interface app_links_web app_links_windows app_links_macos]
- email_validator 2.0.1
- firebase_auth 3.3.19 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.17.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_dynamic_links 4.2.5 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_riverpod 2.0.0-dev.9 [collection flutter meta riverpod state_notifier]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- freezed_annotation 2.0.3 [collection json_annotation meta]
- go_router 3.1.1 [collection flutter flutter_web_plugins logging meta]
- logger 1.1.0
- shared_preferences 2.0.15 [flutter shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]

dev dependencies:
- build_runner 2.1.11 [args async analyzer build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- flutter_lints 2.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- freezed 2.0.3+1 [analyzer build build_config collection meta source_gen freezed_annotation json_annotation]
- json_serializable 6.2.0 [analyzer async build build_config collection json_annotation meta path pub_semver pubspec_parse source_gen source_helper]

transitive dependencies:
- _fe_analyzer_shared 40.0.0 [meta]
- analyzer 4.1.0 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- app_links_macos 1.0.0 [flutter app_links_platform_interface]
- app_links_platform_interface 1.0.2 [flutter plugin_platform_interface]
- app_links_web 1.0.0 [flutter flutter_web_plugins app_links_platform_interface]
- app_links_windows 1.0.0 [flutter app_links_platform_interface]
- args 2.3.1
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- build 2.3.0 [analyzer async convert crypto glob logging meta path]
- build_config 1.0.0 [checked_yaml json_annotation path pubspec_parse yaml]
- build_daemon 3.1.0 [built_collection built_value http_multi_server logging path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.0.9 [analyzer async build crypto graphs logging path package_config pool pub_semver stream_transform yaml]
- build_runner_core 7.2.3 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta path package_config pool timing watcher yaml]
- built_collection 5.1.1
- built_value 8.3.2 [built_collection collection fixnum meta]
- characters 1.2.0
- charcode 1.3.1
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- clock 1.1.0
- code_builder 4.1.0 [built_collection built_value collection matcher meta]
- collection 1.16.0
- convert 3.0.2 [typed_data]
- crypto 3.0.2 [typed_data]
- dart_style 2.2.3 [analyzer args path pub_semver source_span]
- fake_async 1.3.0 [clock collection]
- ffi 2.0.0
- file 6.1.2 [meta path]
- firebase_auth_platform_interface 6.2.7 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 3.3.16 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.4.0 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.6.4 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_dynamic_links_platform_interface 0.2.3+3 [firebase_core flutter meta plugin_platform_interface]
- fixnum 1.0.1
- frontend_server_client 2.1.3 [async path]
- glob 2.0.2 [async collection file path string_scanner]
- graphs 2.1.0 [collection]
- http_multi_server 3.2.0 [async]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- io 1.0.3 [meta path string_scanner]
- js 0.6.4
- json_annotation 4.5.0 [meta]
- lints 2.0.0
- logging 1.0.2
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- mime 1.0.2
- package_config 2.0.2 [path]
- path 1.8.1
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.4 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.0 [ffi flutter path path_provider_platform_interface win32]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.4 [file path platform]
- pub_semver 2.1.1 [collection meta]
- pubspec_parse 1.2.0 [checked_yaml collection json_annotation pub_semver yaml]
- riverpod 2.0.0-dev.9 [collection meta stack_trace state_notifier]
- shared_preferences_android 2.0.12 [flutter shared_preferences_platform_interface]
- shared_preferences_ios 2.1.1 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.1.1 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_macos 2.0.4 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.4 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.1.1 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- shelf 1.3.0 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.2.2 [analyzer async build dart_style glob meta path source_span yaml]
- source_helper 1.3.2 [analyzer collection source_gen]
- source_span 1.8.2 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- state_notifier 0.7.2+1 [meta]
- stream_channel 2.1.0 [async]
- stream_transform 2.0.0
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- timing 1.0.0 [json_annotation]
- typed_data 1.3.1 [collection]
- vector_math 2.1.2
- watcher 1.0.1 [async path]
- web_socket_channel 2.2.0 [async crypto stream_channel]
- win32 2.7.0 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- yaml 3.1.1 [collection source_span string_scanner]
```

</details>

---

**Comment by maheshmnj** [comment]
Hi @mdrideout, Thanks for filing the issue, I am able to reproduce the issue on launching the app from a dynamic link `getinitialLink()`  returns null on IOS.

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.2 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (5 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1237, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.1.0-0.0.pre.1237 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6062e40510 (64 minutes ago), 2022-06-12 23:53:08 -0400
    • Engine revision f80294b57e
    • Dart version 2.18.0 (build 2.18.0-180.0.dev)
    • DevTools version 2.14.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@mdrideout 
Can you take a look at [this](https://github.com/firebase/flutterfire/issues/6913) similar issue and underlying comments / solutions in it and see if any of them help in your case ?

**Comment by mdrideout** [comment]
> Can you take a look at [this](https://github.com/firebase/flutterfire/issues/6913) similar issue and underlying comments / solutions in it and see if any of them help in your case ?

Yes I actually spent quite a long time going through the options and workarounds in that thread without success before opening this issue.

To clarify, I have no issue with the stream listener once the app is already running, only `getInitialLink()`. Also, removing the stream listener does not fix the getInitialLink() issue as suggested by some of the workarounds.

**Comment by darshankawar** [comment]
Thanks for the update. Running firebase_dynamic_links plugin example on iOS and tapping `getInitialLink()` throws `null`.

```
Syncing files to device iPhone 13...
flutter: null
flutter: null
flutter: null
```

/cc @russellwheatley. Similar issue for your reference : https://github.com/firebase/flutterfire/issues/6913

**Comment by TomBeckett** [comment]
@darshankawar Has this been escalated to a developer? 

It seems like its affecting people in live systems and we have ways to [reproduce](https://github.com/firebase/flutterfire/issues/8926#issuecomment-1158682651) is now.

**Comment by darwin-morocho** [comment]
I have the same problem on iOS devices. `getInitialLink()` returns null

**Comment by metajet98** [comment]
I have the same problem, but only in IOS device which isn't install app, open link to app store and install it, then open app from app store (deferred link)
The regular flow for installed app device still working. 
I'm tracing it by logging some information for the production app, cannot testing it because it must through the app store
firebase_dynamic_links: 4.2.6
app_links: 3.2.0

**Comment by salahamassi** [comment]
### before starting describing the problem:

-  `_initialLink` can get just once and if you tried to get it more than once it will be nil, this is the expected behavior as the code shown.
- if you open the through the dynamic link this should trigger `dynamicLinks.onLink` stream and perform any code inside it like a navigation code or something else without having to get the initial link value.

### Problem 

This problem appears because you are almost not added GoogleService-Info.plist to your iOS project

when the app is opened by the dynamic link this method will triggerd 

```objective-c
- (BOOL)application:(UIApplication *)application
    continueUserActivity:(NSUserActivity *)userActivity
      restorationHandler:(nonnull void (^)(NSArray *_Nullable))restorationHandler {
  .............
  [[FIRDynamicLinks dynamicLinks] handleUniversalLink:userActivity.webpageURL
                                           completion:completionBlock];
}
```

and if you try to print the value `FIRDynamicLinks dynamicLinks` instance it will be nil because the instance of the firebase app is still not initialized.

At this stage, the dart code is still not performed so the firebase app instance is still null.

when you add GoogleService-Info.plist the `FLTFirebaseCorePlugin` will initialize the firebase directly from the native code for backward compatibility reasons with old flutter projects.

```objective-c
// Returns a singleton instance of the Firebase Core plugin.
+ (instancetype)sharedInstance {
  static dispatch_once_t onceToken;
  static FLTFirebaseCorePlugin *instance;

  dispatch_once(&onceToken, ^{
    instance = [[FLTFirebaseCorePlugin alloc] init];
    // Register with the Flutter Firebase plugin registry.
    [[FLTFirebasePluginRegistry sharedInstance] registerFirebasePlugin:instance];

    // Initialize default Firebase app, but only if the plist file options
    // exist.
    //  - If it is missing then there is no default app discovered in Dart and
    //  Dart throws an error.
    //  - Without this the iOS/MacOS app would crash immediately on calling
    //  [FIRApp configure] without
    //    providing helpful context about the crash to the user.
    //
    // Default app exists check is for backwards compatibility of legacy
    // FlutterFire plugins that call [FIRApp configure]; themselves internally.
    FIROptions *options = [FIROptions defaultOptions];
    if (options != nil && [FIRApp allApps][@"__FIRAPP_DEFAULT"] == nil) {
      [FIRApp configureWithOptions:options];
    }
  });

  return instance;
```
}

and when the `dart Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform)` called the firebase app instance will renitialized again this time using the firebase options from dart code.
```objective-c
- (void)initializeAppAppName:(nonnull NSString *)appName
        initializeAppRequest:(nonnull PigeonFirebaseOptions *)initializeAppRequest
                  completion:(nonnull void (^)(PigeonInitializeResponse *_Nullable,
                                               FlutterError *_Nullable))completion {
     .............
     
    [FIRApp configureWithName:appNameIos options:options];

  completion([self initializeResponseFromFIRApp:[FIRApp appNamed:appNameIos]], nil);
}
```


### Conclusion: 
This is not a firebase dynamic link issue this is a firebase core issue and there are two workarounds for this issue

1- simply add GoogleService-Info.plist to your ios project.
2- add some delay before calling  ```[[FIRDynamicLinks dynamicLinks] handleUniversalLink];``` 
method, so we make sure that the firebase app is reinitialized from the dart code with firebase options.

**Comment by mdrideout** [comment]
> 1- simply add GoogleService-Info.plist to your ios project.
2- add some delay before calling [[FIRDynamicLinks dynamicLinks] handleUniversalLink];

This does not work for me, I set up a fresh skeleton app with up-to-date GoogleService-Info.plist and it does not work. I have tried adding a delay, it does not work. Firebase is initialized correctly and I have no issues with any other services or other features of Dynamic Links.

**Comment by salahamassi** [comment]
> > 1- simply add GoogleService-Info.plist to your ios project.
> > 2- add some delay before calling [[FIRDynamicLinks dynamicLinks] handleUniversalLink];
> 
> This does not work for me, I set up a fresh skeleton app with up-to-date GoogleService-Info.plist and it does not work. I have tried adding a delay, it does not work. Firebase is initialized correctly and I have no issues with any other services or other features of Dynamic Links.

- The App links library can get your initial link because it doesn't have to handle the link using the firebase dynamic link.
- You have to add delay on the native code after creating your own fork from the flutter fire plugins.
- make sure you add the GoogleService-Info.plist on the right way using Xcode 


Flutter app output before adding GoogleService-Info.plist file

```
Running Xcode build...
Xcode build done.                                           42.5s
8.15.0 - [Firebase/Core][I-COR000012] Could not locate configuration file: 'GoogleService-Info.plist'.
Debug service listening on ws://127.0.0.1:60000/4wGNiaK8dl8=/ws
Syncing files to device iPhone 13 Pro...
8.15.0 - [Firebase/Core][I-COR000005] No app has been configured yet.
8.15.0 - [Firebase/Core][I-COR000005] No app has been configured yet.

```

A video trying to open the app directly from the dynamic link from the app terminated state
 
https://user-images.githubusercontent.com/17902030/176045347-8554fbf9-e9ae-4858-a953-0c69f4336c5f.mp4


log after adding GoogleService-Info.plist file

```
Running Xcode build...
Xcode build done.                                           12.6s
8.15.0 - [Firebase/Core][I-COR000005] No app has been configured yet.
Debug service listening on ws://127.0.0.1:60249/sX6NGK9eheU=/ws
Syncing files to device iPhone 13 Pro...

```

A video trying to open the app directly from the dynamic link from the app terminated state

https://user-images.githubusercontent.com/17902030/176047308-48ff91cb-bb39-499a-a56b-8b40e4eb34ad.mp4


### Flutter doctor

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.3.1 21E258 darwin-x64, locale en-PS)
[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0-rc1)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.68.1)
[✓] Connected device (4 available)
    ! Error: iPhone is not connected. Xcode will continue when iPhone is connected. (code -13)
    ! Error: Salah Iphone 12 is busy: Preparing the watch for development via Salah Iphone 12. Xcode will continue when Salah Iphone 12 is finished. (code -10)
    ! Error: Ibrahim’s iPhone is not connected. Xcode will continue when Ibrahim’s iPhone is connected. (code -13)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
```

</details>

---


### Flutter dependencies

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.1
Flutter SDK 3.0.1
firebase_dynamic_links_example 0.0.0

dependencies:
- firebase_core 1.18.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_dynamic_links 4.2.6 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- url_launcher 6.1.4 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

dependency overrides:
- firebase_core 1.18.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_platform_interface 4.4.1 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_dynamic_links 4.2.6 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- firebase_dynamic_links_platform_interface 0.2.3+4 [firebase_core flutter meta plugin_platform_interface]

transitive dependencies:
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.16.0
- fake_async 1.3.0 [clock collection]
- firebase_core_web 1.6.5 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- js 0.6.4
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- path 1.8.1
- plugin_platform_interface 2.1.2 [meta]
- sky_engine 0.0.99
- source_span 1.8.2 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.0 [flutter plugin_platform_interface]
- url_launcher_web 2.0.12 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- vector_math 2.1.2```

</details>

---

**Comment by russellwheatley** [comment]
I've just tested this on my iOS device (required) and ran on `--release` (also required). My initial link was not null. However, I did have to use manual installation of the google services file (`GoogleService-Info.plist`). We are looking to implement a write of this file when you use `flutterfire configure` to get your Firebase config. [See here](https://github.com/invertase/flutterfire_cli/pull/70) as it is become clear that initializing from dart isn't feasible for all Firebase products which we don't have control over their underlying implementation. I'm going to get this prioritised.

**Comment by JakeHadley** [comment]
This is really something that I've personally torn my hair out over. No where explicitly in the docs when setting up the basics of firebase in flutter does it say I need to add the GoogleService-Info.plist. But when I start adding plugins and running and testing my app, I get logs in xcode saying that the file wasn't found. Well, yeah. No one said I needed it. The docs should at least mention that it may be needed, and then provide a link to show the correct way of adding it to your project. 

> We are looking to implement a write of this file when you use flutterfire configure to get your Firebase config.

This is what I expected to happen when running `flutterfire configure` and was quite confused when it didn't generate the file, and I was left thinking it wasn't necessary anymore (ignorant of me, I know). But since this isn't a feature already, it would be quite wonderful to have it included. Good to see it's in a preview build.

**Comment by russellwheatley** [comment]
FlutterFire CLI now supports writing the `GoogleService-Info.plist` in the Runner when you run `flutterfire configure` which should solve the issue. Please update FlutterFire CLI by running `dart pub global activate flutterfire_cli` first. Thanks.
